<a href="https://colab.research.google.com/github/Everlead/Hamoye-quiz-code/blob/master/Hamoye_quiz_code.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# covention for importing numpy
import pandas as pd


In [9]:
#create URL
url = 'https://raw.githubusercontent.com/WalePhenomenon/climate_change/master/fuel_ferc1.csv'
fuel_data = pd.read_csv(url)

In [10]:
pd.read_csv(url)

record_id  ...  fuel_cost_per_mmbtu
0        f1_fuel_1994_12_1_0_7  ...                1.121
1       f1_fuel_1994_12_1_0_10  ...                1.120
2        f1_fuel_1994_12_2_0_1  ...                1.650
3        f1_fuel_1994_12_2_0_7  ...                1.970
4       f1_fuel_1994_12_2_0_10  ...                2.570
...                        ...  ...                  ...
29518  f1_fuel_2018_12_12_0_13  ...                9.030
29519   f1_fuel_2018_12_12_1_1  ...                6.950
29520  f1_fuel_2018_12_12_1_10  ...                8.990
29521  f1_fuel_2018_12_12_1_13  ...                1.110
29522  f1_fuel_2018_12_12_1_14  ...               11.680

[29523 rows x 11 columns]

In [11]:
#import pandas and numpy
import pandas as pd
import numpy as np

Question 1

In [12]:
#identity matrix
np.identity(3)

array([[1., 0., 0.],
       [0., 1., 0.],
       [0., 0., 1.]])

Question 2


In [48]:
#the second and third lowest correlation with the fuel cost per unit burned
fuel_data.corr().loc['fuel_cost_per_unit_burned'].abs().sort_values()

fuel_cost_per_mmbtu             0.000437
fuel_mmbtu_per_unit             0.010034
fuel_cost_per_unit_delivered    0.011007
report_year                     0.013599
fuel_qty_burned                 0.018535
utility_id_ferc1                0.037863
fuel_cost_per_unit_burned       1.000000
Name: fuel_cost_per_unit_burned, dtype: float64

Second = fuel_mmbtu_per_unit             0.010034

Third = fuel_cost_per_mmbtu             0.000437


Question 3

In [16]:
#calculating the skewness of fuel quatity burned
fuel_data['fuel_qty_burned'].skew()


15.851495469109503

In [18]:
#calculating the kurtosis of the fuel quantity burned
fuel_data['fuel_qty_burned'].kurt()

651.3694501337732

Question 4

In [19]:
#standard deviation
std_deviation = fuel_data['fuel_mmbtu_per_unit'].std()
print('standard deviation is: ' + str(std_deviation))

standard deviation is: 10.600220307806886


In [23]:
#percentile
percentile = fuel_data['fuel_mmbtu_per_unit'].quantile(q=0.75, interpolation='linear')
print('75% quantile is:' + str(percentile))

75% quantile is:17.006


Question 5

In [24]:
#the lowest average fuel cost per unit burned
fuel_data.groupby('fuel_type_code_pudl')['fuel_cost_per_unit_burned'].mean().sort_values()

fuel_type_code_pudl
gas          13.659397
other        18.253856
waste        19.518122
coal         67.421830
oil         168.877086
nuclear    4955.157002
Name: fuel_cost_per_unit_burned, dtype: float64

Queston 6

A python list extend fuction through the value to be added to the list, A,extend(B)


Question 7

In [37]:
#using the isnull fuction to get the null values and the sum fuction to know the total number of the null values
null = fuel_data['fuel_unit'].isnull().sum()

total_value = fuel_data['record_id'].count()

percentage = (null/total_value) * 100
percentage

0.609694136774718

In [38]:
#using the non null value of the feature fuel_unit
total_value1 = fuel_data['fuel_unit'].count()
percentage1 = (null/total_value1) * 100
percentage1

0.6134342091810653

Question 8

In [40]:
#the percentage change in the fuel cost per unit burned in 1998 to 1994
yearly_cost = fuel_data.groupby(['report_year', 'fuel_type_code_pudl'])['fuel_cost_per_unit_burned'].sum()
year_1994, year_1998 = yearly_cost.loc[[1994, 1998], 'coal']
percentage_change = ((year_1998 - year_1994)/year_1994) * 100
percentage_change

-20.56765451826049

Question 9

In [46]:
#the highest average fuel cost per unit delivered
fuel_data.groupby('report_year')['fuel_cost_per_unit_delivered'].mean().sort_values(ascending=True)

report_year
1999       25.551627
1995       32.735269
2006       38.657484
2005       41.438184
2007       43.325023
2017       46.196861
2002       47.594361
2003       55.663493
2008       58.588197
2011       59.774667
2001       60.050396
2012       60.994502
1994       63.636060
2010       91.862105
2016      103.901761
2004      139.524275
2013      172.307591
2014      192.737183
1998      287.154420
2015      326.535511
2018      499.269966
2009      652.694163
2000      985.362877
1996     9196.705948
1997    11140.197239
Name: fuel_cost_per_unit_delivered, dtype: float64

Question 10


Categorical and mode imputation